src/2-price-update.rain

```diff
@@ -0,0 +1,69 @@
+/*op meta*/
+@0x47ed85f917e187757bff09371cedcf5c0eb277c27e4673feb2d3cc040c66c993
+
+/*ob meta*/
+@0x56ffc3fc82109c33f1e1544157a70144fc15e7c6e9ae9c65a636fd165b1bc51c
+/* Calculate IO */
+
+/* refuse any counterparties other than named */
+allowed-counterparty: {{ counterparty }},
+: ensure(eq(allowed-counterparty counterparty-address())),
+
+/* 1 hour delay on new batches = 3600 seconds */
+batch-start-info-k: order-hash(),
+batch-start-info: get(batch-start-info-k),
+batch-start-time: decode-256<32 32>(batch-start-info),
+: ensure(gt(now() int-add(batch-start-time 3600))),
+
+/* get info about current batch */
+batch-index batch-remaining: call<2 2>(0),
+
+/*
+ * Exponential growth
+ * A * (1 + r)^x
+ * A = initial ratio
+ * r = % increase per batch
+ * x = batch index
+ */
+io-ratio: decimal18-mul({{ ratio }} decimal18-pow-int(101e16 batch-index)),
```

Comment:
we discussed removing the {{ }} syntax as it breaks vs code and makes it impossible to review values in PRs

Reply:
Resolved.